src/Nav.js

```diff
@@ -1,359 +1,233 @@
+     * the ARIA authoring practices for tabs:
+     * https://www.w3.org/TR/2013/WD-wai-aria-practices-20130307/#tabpanel
+     */
+    role: PropTypes.string,
+
+    /**
+     * Apply styling an alignment for use in a Navbar. This prop will be set
+     * automatically when the Nav is used inside a Navbar.
+     */
+    navbar: PropTypes.bool,
+
+    componentClass: elementType,
+
+    /** @private */
+    onKeyDown: PropTypes.func
+  };
+
+  static defaultProps = {
+    justify: false,
+    fill: false,
+    componentClass: 'ul'
+  };
+
+  static getDerivedStateFromProps(
+    { activeKey, getControlledId, getControllerId, role, onSelect },
+    prevState
+  ) {
+    return {
+      ...prevState,
+      navContext: {
+        role, // used by NavLink to determine it's role
+        onSelect,
+        activeKey,
+        getControlledId: getControlledId || noop,
+        getControllerId: getControllerId || noop
+      }
+    };
   }
 
-  getActiveProps() {
-    const tabContainer = this.context.$bs_tabContainer;
+  constructor(...args) {
+    super(...args);
 
-    if (tabContainer) {
-      warning(
-        this.props.activeKey == null && !this.props.activeHref,
-        'Specifying a `<Nav>` `activeKey` or `activeHref` in the context of ' +
-          'a `<TabContainer>` is not supported. Instead use `<TabContainer ' +
-          `activeKey={${this.props.activeKey}} />\`.`
-      );
+    this.state = { navContext: null };
+  }
 
-      return tabContainer;
-    }
+  componentDidUpdate() {
+    if (!this._needsRefocus || !this.listNode) return;
 
-    return this.props;
+    let activeChild = qsa(this.listNode, `.nav-link.active`).pop();
+    if (activeChild) activeChild.focus();
   }
 
   getNextActiveChild(offset) {
-    const { children } = this.props;
-    const validChildren = children.filter(
-      child => child.props.eventKey != null && !child.props.disabled
-    );
-    const { activeKey, activeHref } = this.getActiveProps();
+    if (!this.listNode) return null;
+    const itemSelector = '.nav-link:not(.disabled)';
 
-    const activeChild = ValidComponentChildren.find(children, child =>
-      this.isActive(child, activeKey, activeHref)
-    );
-
-    // This assumes the active child is not disabled.
-    const activeChildIndex = validChildren.indexOf(activeChild);
-    if (activeChildIndex === -1) {
-      // Something has gone wrong. Select the first valid child we can find.
-      return validChildren[0];
-    }
+    let items = qsa(this.listNode, itemSelector);
+    let activeChild = qsa(this.listNode, `${itemSelector}.active`).pop();
 
-    let nextIndex = activeChildIndex + offset;
-    const numValidChildren = validChildren.length;
-
-    if (nextIndex >= numValidChildren) {
-      nextIndex = 0;
-    } else if (nextIndex < 0) {
-      nextIndex = numValidChildren - 1;
-    }
-
-    return validChildren[nextIndex];
+    return items[items.indexOf(activeChild) + offset];
```

Comment:
suppose there's no active child. then this will give the item with index "offset - 1". is that right?